Assets/Scripts/UI/SettingsMenu.cs

```diff
@@ -0,0 +1,145 @@
+using System.Collections.Generic;
+using FrameWork;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Audio;
+using Slider = UnityEngine.UI.Slider;
+
+namespace UI
+{
+    public class SettingsMenu : MonoBehaviour
+    {
+        [SerializeField] private AudioMixer audioMixer;
+        [SerializeField] private TMP_Dropdown graphicsDropdown;
+        [SerializeField] private TMP_Dropdown resolutionsDropdown;
+        [SerializeField] private Slider slider;
+        [SerializeField] private Slider progressSlider;
+        [SerializeField] private bool customSettings;
+        [SerializeField] private SceneSwitcher _sceneSwitcher;
+
+        private Resolution[] _resolution;
+
+        private void Awake()
+        {
+            if (progressSlider != null) progressSlider.value = _sceneSwitcher.Progress;
+            LoadSettings();
+        }
+
+        #region Loading
+        /// <summary>
+        /// This loads all the custom settings, if there are no custom settings it will run default settings.
+        /// </summary>
+        public void LoadSettings()
+        {
+            PopulateResolutions();
+            customSettings = PlayerPrefs.GetInt("CustomSettings", customSettings ? 1 : 0) == 1;
+            if (!customSettings)
+            {
+                SetHighestQuality();
+                PlayerPrefs.DeleteAll();
+            }
+            float volume = PlayerPrefs.GetFloat("Volume");
+            int quality = PlayerPrefs.GetInt("Quality", QualitySettings.GetQualityLevel());
+            bool fullscreen = PlayerPrefs.GetInt("Fullscreen", Screen.fullScreen ? 1 : 0 ) == 1;
+            int resolution = PlayerPrefs.GetInt("Resolution", GetDefaultResolution());
+            
+            SetVolume(volume);
+            SetQuality(quality);
+            SetFullscreen(fullscreen);
+            SetResolution(resolution);
+            
+        }
+        #endregion
+
+        #region Settings
+                public void ApplySettings()
```

Comment:
```suggestion
        /// <summary>
        /// 
        /// </summary>
        public void ApplySettings()
```
Waarom hier extra tab? Je mist een summary hierzo.